README.md

```diff
@@ -1,2 +1,33 @@
 # control-plane
-Go implementation of data-plane-api
+This repository contains a Go-based implementation of an API server that
+implements the discovery service APIs defined in
+[data-plane-api](https://github.com/envoyproxy/data-plane-api).
+
+## Scope
+
+Due to the variety of platforms out there, there is no single
+control plane implementation that can satisfy everyone's needs.  Hence this
+code base does not attempt to be a full scale control plane for a fleet of
+Envoy proxies. Instead, it provides infrastructure that is shared by
+multiple different control plane implementations. The components provided
+by this library are:
+
+* _API Server:_ A generic gRPC based API server that implements xDS APIs as defined
+  in the
+  [data-plane-api](https://github.com/envoyproxy/data-plane-api). The API
+  server is responsible for pushing configuration updates to
+  Envoys. Consumers should be able to import this go library and use the
+  API server as is, in production deployments.
+
+* _Configuration Cache:_ The library will cache Envoy configurations in
+memory in an attempt to provide fast response to consumer Envoys. It is the
+responsibility of the consumer of this library to populate the cache as
+well as invalidate it when necessary. The cache will be keyed based on a
+pre-defined hash function whose keys are based on the
+[Node](https://github.com/envoyproxy/data-plane-api/blob/master/api/base.proto).
```

Comment:
Nit: link to a specific line number and hash as best practice.

Reply:
This was intentional, as I didn't know if the line numbers would remain same, as we change the protos. Once we have documentation, we could link to the Node message's documentation. WDYT?

Reply:
You can just link to a specific hash (GH even has some  click on line, click on ... on LHS, select "Copy Permalink"), e.g. https://github.com/envoyproxy/data-plane-api/blob/d4988844024d0bcff4bcd030552eabe3396203fa/api/base.proto#L26.

Reply:
You can do a range of lines too:

https://github.com/envoyproxy/data-plane-api/blob/d4988844024d0bcff4bcd030552eabe3396203fa/api/base.proto#L26-L36

Reply:
Ah I didn't know the range of lines trick. Thats neat.. updated

Reply:
it's newish. we're pretty enamored with it.